src/components/table.js

```diff
@@ -214,11 +216,19 @@ export class GridTable extends Component {
   constructor(props) {
     super(props)
     this.state = { scrollbarSize: 0 }
-    this.grid = createRef()
+    this.header = createRef()
+    this.body = createRef()
   }
 
   componentDidMount() {
-    this.grid.current.measureAllCells()
+    this.body.current.measureAllCells()
+  }
+
+  componentDidUpdate(prevProps) {
+    if (_.map('width', prevProps.columns) !== _.map('width', this.props.columns)) {
```

Comment:
This check will always be true, because it's checking references, not values. Also, this could be somewhat expensive if the number of columns is very large. Perhaps we can just delegate this responsibility to the parent? E.g. expose a method to recompute both, and have the parent take a ref and call the method when things resize (since we know for for sure at that level).